CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs

```diff
@@ -1,13 +1,14 @@
 ﻿using System;
+using System.Collections.Generic;
 
 namespace CollisionDetectionSystem
 {
-	public delegate void DataDel(TransponderData data);
+	public delegate void DataDel( List<TransponderData> data);
 
 	public interface ITransponderReceiver
 	{
-		void ReceiveData(TransponderData data);
-		void PrepareDataForPost(TransponderData data);
+		void ReceiveData(List<TransponderData> data);
+		void PrepareDataForPost( List<TransponderData> data);
 		event DataDel PostDataEvent;
```

Comment:
yeah i kinda agree, it seems more logical to send out all the info on airplanes who's info we received in the last second then send out every time you get one. I don't really know though.  


    On Friday, November 20, 2015 10:23 AM, speppers <notifications@github.com> wrote:
 

 In CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:>  
>  	public interface ITransponderReceiver
>  	{
> -		void ReceiveData(TransponderData data);
> -		void PrepareDataForPost(TransponderData data);
> +		void ReceiveData(List<TransponderData> data);
> +		void PrepareDataForPost( List<TransponderData> data);
>  		event DataDel PostDataEvent;
If a transponder is tracking multiple planes doesn't it send data of all in range planes at a set interval at the same time? Other wise each data sent from the transponder would be off time wise by a couple of milliseconds per intruder so if there were 1000 intruders the last intruder to get sent would be significantly different time wise then the first?…Sent from my iPhone On Nov 20, 2015, at 9:57 AM, dsteeve <notifications@github.com> wrote: In CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs: > > public interface ITransponderReceiver > { > - void ReceiveData(TransponderData data); > - void PrepareDataForPost(TransponderData data); > + void ReceiveData(List<TransponderData> data); > + void PrepareDataForPost( List<TransponderData> data); > event DataDel PostDataEvent; Hey Fred, I don't understand why we want a List here. This TransponderReceiver is suppose to receive just one transponderData at a time. This in the real world would be the transponder getting one ping from a plane. It would not get a list. I will checkout out your branch and try to understand a bit more. — Reply to this email directly or view it on GitHub.—
Reply to this email directly or view it on GitHub.